addons/l10n_ar/models/account_move.py

```diff
@@ -170,17 +180,13 @@ def _post(self, soft=True):
         ar_invoices = self.filtered(lambda x: x.company_id.country_id.code == "AR" and x.l10n_latam_use_documents)
         for rec in ar_invoices:
             rec.l10n_ar_afip_responsibility_type_id = rec.commercial_partner_id.l10n_ar_afip_responsibility_type_id.id
-            if rec.company_id.currency_id == rec.currency_id:
-                rec.l10n_ar_currency_rate = 1.0
-            elif not rec.l10n_ar_currency_rate:
-                rec.l10n_ar_currency_rate = rec.currency_id._convert(
-                    1.0, rec.company_id.currency_id, rec.company_id, rec.date, round=False)
 
         # We make validations here and not with a constraint because we want validation before sending electronic
         # data on l10n_ar_edi
         ar_invoices._check_argentinian_invoice_taxes()
         posted = super()._post(soft)
-        posted._set_afip_service_dates()
+        ar_invoices._set_afip_rate()
+        ar_invoices._set_afip_service_dates()
```

Comment:
@jco-odoo Doesn't this need to be the intersection beween `posted` and `ar_invoices`?

Reply:
Most of the _post inherits do not.  Future move posting does not apply here?

Reply:
> Most of the _post inherits do not.

That is probably a mistake.

For instance if you prepared an invoice for in the future, setting it to auto-post by calling `_post(soft=True)`, you (very) most likely want to call `_set_afip_rate` with the latest rates that don't exist yet. So you want to compute it when it will actually be posted for real, not when you decided that it will be posted eventually.